library/src/main/kotlin/de/adesso/junitinsights/tools/ReportCreator.kt

```diff
@@ -27,25 +27,26 @@ object ReportCreator {
      */
     private fun processClassEvents(events: List<Event>): TestClass {
         val beforeAll = events[0].timeStamp.time
-        val spring = if (events[1].name.equals("context refreshed")) events[1].timeStamp.time else events[0].timeStamp.time
+        val spring = if (events[1].name == "context refreshed") events[1].timeStamp.time else events[0].timeStamp.time
         val afterAll = events.last().timeStamp.time
         val eventsGroupedByMethods = groupEventsByMethod(events)
         val methods = eventsGroupedByMethods.map { methodEvents -> processMethodEvents(methodEvents) }
 
         if (methods.isEmpty())
-            return TestClass(events.last().className, methods, 0, 0, 0, 0, 0, 0, 0)
+            return TestClass(events.last().className, events.first().timeStamp.time, methods, 0, 0, 0, 0, 0, 0, 0)
 
         var between = 0L
         for (i in 1..methods.lastIndex)
             between += methods[i].timestampBefore - methods[i - 1].timestampAfter
 
         return TestClass(
                 events.last().className,
+                events.first().timeStamp.time,
                 methods,
                 methods[0].timestampBefore - spring,
-                methods.map { method -> method.before }.sum(),
-                methods.map { method -> method.exec }.sum(),
-                methods.map { method -> method.after }.sum(),
+                methods.asSequence().map { method -> method.before }.sum(),
+                methods.asSequence().map { method -> method.exec }.sum(),
+                methods.asSequence().map { method -> method.after }.sum(),
```

Comment:
What is the `.asSequence()` for? It seems to work fine without it.

Reply:
Sequences in Kotlin use lazy loading to do their calculations (see [this blogpost](https://blog.kotlin-academy.com/effective-kotlin-use-sequence-for-bigger-collections-with-more-than-one-processing-step-649a15bb4bf?gi=fea8d965e8a5)).

There are to be preferred when doing more than one step on a collection of things for performance and portability reasons. This is why IntelliJ hints this addition.

Reply:
Okay, I don't think this is necessary and it does not really belong into this PR, but it improves the performance a little bit so I think it can stay.

My IDE does not suggest this change, however which is kind of odd, because I am using IntelliJ, too.

Reply:
That is indeed odd. I normally apply reasonable suggestions immediately, so I'm not sure why I didn't put it in to begin with. Maybe the suggestion is in the newest version of IntelliJ